lib/companies/pending_changes.ex

```diff
@@ -55,9 +55,15 @@ defmodule Companies.PendingChanges do
     end
   end
 
-  defp apply_changes("create", changeset), do: Repo.insert(changeset)
-  defp apply_changes("delete", record), do: Repo.delete(record)
-  defp apply_changes("update", changeset), do: Repo.update(changeset)
+  defp apply_changes("create", changeset, _change_id), do: Repo.insert(changeset)
+
+  defp apply_changes("delete", record, change_id) do
+    record
+    |> Ecto.Changeset.cast(%{is_deleted: true, deleted_via_id: change_id}, [:is_deleted, :deleted_via_id])
```

Comment:
I don't know that this is necessary. I think we can ditch the `"delete"` action in favor of setting `deleted: true` on the delete route?